### DISCUSSION TITLE
how to update customSettings programmable

### DISCUSSION THREAD

**Discussion by vitalybrandes** [OrchardCMS/OrchardCore#7213] [Q&A] [answered]
I create custom settings based on youtube video in documentation! Perfecto!!!

@deanmarcussen , can you please navigate me how can i update this custom setting programmable?
i need to update textfiled in the settings.

**Comment by vitalybrandes** [comment]
anyone, please ,help .....

**Reply by Skrypt** [reply]
In YesSQL an IEntity always contains a Property field. We use this field to store the CustomSettings.
You can see an example of that with `public Task UpdateSiteSettingsAsync(ISite site)` where we save an ISite which inherits from IEntity. So basically `IEntity.Properties[PropertyName] = value`;

**Reply by vitalybrandes** [reply]
Thank you, will check!

**Comment by vitalybrandes** [comment, marked as answer]
If some one need , what i did to update CustomSettings:

```C#
var contentTypeDefinition = _customSettingsService.GetSettingsType("SettingsType");

if (!await _customSettingsService.CanUserCreateSettingsAsync(contentTypeDefinition))
{
    return false;
}

var container = await _siteService.LoadSiteSettingsAsync();

var Settings = container.As<ContentItem>("SettingsType");
if (Settings != null)
{
    Settings.Alter<ContentPart>("Settings", a =>
    {
        a.Alter<TextField>("Yourtextfiled", a => a.Text = YourVaule);
    });
}

container.Properties[nameof(Settings)] = JObject.FromObject(Settings);

await _siteService.UpdateSiteSettingsAsync(container)
```

**Comment by hishamco** [comment]
@vitalybrandes all the time the guys point you to the solution and you write the complete code and mark it as answer ;)

**Reply by vitalybrandes** [reply]
Guys, you are the best!! No doubt,
Just from my experience, when someone looking for solution, it's nicer to get all the code.
Nothing personal, just trying to help.

**Reply by vitalybrandes** [reply]
P.s I found the solution with your help @Skrypt , thank you!!!! 😉

**Reply by hishamco** [reply]
Just joking .. code to see a complete code

Thanks

**Comment by weirdyang** [comment]
Is it possible to access customSettings in workflows?
